app/utils/Strings.scala

```diff
@@ -38,6 +38,37 @@ object Strings {
     Random.alphanumeric.filter(_.isDigit).take(length).mkString
   }
 
+  def decimalMax12(): String =
+    Seq(
+      Gen.choose(1, 9).sample.getOrElse(1).toString,
+      Gen.choose(1, 9).sample.getOrElse(2).toString,
+      Gen.choose(1, 9).sample.getOrElse(3).toString,
+      Gen.choose(1, 9).sample.getOrElse(4).toString,
+      Gen.choose(1, 9).sample.getOrElse(5).toString,
+      Gen.choose(1, 9).sample.getOrElse(6).toString,
+      Gen.choose(1, 9).sample.getOrElse(7).toString,
+      Gen.choose(1, 9).sample.getOrElse(8).toString,
+      Gen.choose(1, 9).sample.getOrElse(9).toString,
+      Gen.choose(1, 9).sample.getOrElse(3).toString,
+      Gen.choose(1, 9).sample.getOrElse(5).toString,
+    ).mkString
+
+  def numeric8(): String = {
+    val first  = Gen.choose(1, 2).sample.getOrElse(1)
+    val second = Gen.choose(0, 9).sample.getOrElse(9)
+
+    Seq(
+      first.toString,
+      second.toString,
+      first.toString,
+      second.toString,
+      "0",
+      Gen.choose(1, 9).sample.getOrElse(5).toString,
+      "0",
+      Gen.choose(1, 9).sample.getOrElse(7).toString,
+    ).mkString
+  }
```

Comment:
Not sure I'm a fan of the val names and def name in this. It isn't very descriptive. If I called numeric 8 I'd expect to get something 8 digits long all random.

Reply:
I agree with the naming not being descriptive but It matches the name for the pattern found in the XSD file.

Reply:
If it's just 8 numbers then, why is the generator different for most of the digits?

Reply:
It's to follow the patten defined in the regex. Acceptable values are different for each part of the string.

Reply:
fine. I don't like it though.